### PR TITLE
File Upload dynamic progress bar

### PR DESCRIPTION
**Pull request by ChazUK**

### CODE REVIEW

File: src/app/core/services/bulk-upload.service.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
-import { BehaviorSubject, Observable } from 'rxjs';
-import { ErrorDetails } from '@core/model/errorSummary.model';
-import { EstablishmentService } from '@core/services/establishment.service';
-import { FormGroup } from '@angular/forms';
 import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
 import { Injectable } from '@angular/core';
-import { map } from 'rxjs/operators';
+import { FormGroup } from '@angular/forms';
 import { PresignedUrlResponse, UploadFile } from '@core/model/bulk-upload.model';
+import { ErrorDetails } from '@core/model/errorSummary.model';
```
Comment (omarCreativeDev) [inline src/app/core/services/bulk-upload.service.ts]: are these imports not alphabetised anymore?

File: src/app/features/bulk-upload/files-upload/files-upload.component.html [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
   <app-selected-files-list *ngIf="!filesUploaded"></app-selected-files-list>
 
-  <div *ngIf="!filesUploading && !filesUploaded">
```
Comment (omarCreativeDev) [inline src/app/features/bulk-upload/files-upload/files-upload.component.html]: ah good spot removing `!filesUploaded` flag as of course we want to allow the user to re-upload if needed

File: src/app/features/bulk-upload/files-upload/files-upload.component.ts [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
-import { forkJoin, Observable } from 'rxjs';
 import { AbstractControl, FormBuilder, FormGroup, Validators } from '@angular/forms';
-import { mergeMap, take } from 'rxjs/operators';
-import { CustomValidators } from '@shared/validators/custom-form-validators';
 import { UploadFile } from '@core/model/bulk-upload.model';
+import { BulkUploadService } from '@core/services/bulk-upload.service';
+import { ErrorSummaryService } from '@core/services/error-summary.service';
+import { CustomValidators } from '@shared/validators/custom-form-validators';
+import { combineLatest, forkJoin, Observable, Subscription } from 'rxjs';
+import { map, take, tap } from 'rxjs/operators';
```
Comment (omarCreativeDev) [inline src/app/features/bulk-upload/files-upload/files-upload.component.ts]: same here, are these imports not alphabetised anymore?

File: src/assets/scss/components/_progress.scss [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  &[value] {
+    appearance: none;
+
+    border: none;
+    height: 30px;
+
+
+    &::-webkit-progress-bar {
+      background: transparent;
+      border: 2px solid govuk-colour("grey-1");
```
Comment (omarCreativeDev) [inline src/assets/scss/components/_progress.scss]: is `grey-1` a color? can we use a variable and store it in colors.scss instead?

File: src/assets/scss/components/_progress.scss [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    }
+
+    /* IE10 */
+    color: #00823b;
```
Comment (omarCreativeDev) [inline src/assets/scss/components/_progress.scss]: can we use a variable and store this hex in colors.scss

File: src/app/core/services/bulk-upload.service.ts (hunk 1 above)
Comment (ChazUK) [reply src/app/core/services/bulk-upload.service.ts]: They are ordered by their import location

File: src/assets/scss/components/_progress.scss (hunk 5 above)
Comment (ChazUK) [reply src/assets/scss/components/_progress.scss]: Updated

**Comment by ChazUK** [comment]
[(3) Bulk Upload - Upload Progress Bar](https://trello.com/c/ohRw9ljm/70-3-bulk-upload-upload-progress-bar)

**Pull request merged**